### PR TITLE
LoggingTrait Documentation with Example

### PR DESCRIPTION
**Pull request by davisgomes**
For the new logging trait changes, we are documenting the API for the LoggingTrait CRD.

### CODE REVIEW

File: content/en/docs/reference/API/OAM/LoggingTrait.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
+title: LoggingTrait Custom Resource Definition
+linkTitle: LoggingTrait CRD
+weight: 2
+draft: false
+---
+The LoggingTrait custom resource contains the configuration for an additional logging sidecar with a custom image and fluentd config.
```
Comment (markxnelson) [inline content/en/docs/reference/API/OAM/LoggingTrait.md:7]: I think Fluentd needs a capital

Comment (markxnelson) [reply content/en/docs/reference/API/OAM/LoggingTrait.md:7]: probably should say "configuration file" not "config" too

File: content/en/docs/reference/API/OAM/LoggingTrait.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+Here is a sample ApplicationConfiguration that includes a LoggingTrait.
```
Comment (markxnelson) [inline content/en/docs/reference/API/OAM/LoggingTrait.md:8]: put ApplicationConfiguration and LoggingTrait in backticks

Comment (davisgomes) [reply content/en/docs/reference/API/OAM/LoggingTrait.md]: Thanks, Fixed this throughout the doc

File: content/en/docs/reference/API/OAM/LoggingTrait.md (hunk 1 above)
Comment (davisgomes) [reply content/en/docs/reference/API/OAM/LoggingTrait.md]: Fixed these two comments throughout the doc

File: content/en/docs/reference/API/OAM/LoggingTrait.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+#### LoggingTraitSpec
+LoggingTraitSpec specifies the desired state of an ingress trait.
+
+| Field | Type | Description | Required
+| --- | --- | --- | --- |
+| `loggingConfig` | map[string]string | A  of the Fluentd configuration file name and the configuration file details. | Yes |
+| `loggingImage` | string | The name of the custom Fluentd image. | Yes |
+| `workloadRef` | [WorkloadReference](#workloadReference) | The name of the custom Fluentd image. | Yes |
+
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/OAM/LoggingTrait.md:75]: Is it correct that the description of `loggingImage` is the same as the description of `workloadRef`?

File: content/en/docs/reference/API/OAM/LoggingTrait.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+#### WorkloadReference
+WorkloadReference specifies the desired workload information.
+
+| Field | Type | Description | Required
+| --- | --- | --- | --- |
+| `apiVersion` | string | API Version for the desired workload | Yes |
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/OAM/LoggingTrait.md:81]: Version for -> version of
workload -> workload. (period at the end of the sentence.) Fix the next one too, please.

File: content/en/docs/reference/API/OAM/LoggingTrait.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+| `kind` | string | Kind of the desired workload | Yes |
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/OAM/LoggingTrait.md:82]: Kind of the desired workload -> (this wording seems awkward to me) -> Kind of desired workload. (is that ok?)

**Review by rosemarymarano — COMMENTED** [review]
A few edits

File: content/en/docs/reference/API/OAM/LoggingTrait.md [hunk 6]
```diff
@@ -62,10 +62,10 @@ In this example, the `logging-stdout` container will run the image given in the
 | `apiVersion` | string | `oam.verrazzano.io/v1alpha1` | Yes |
 | `kind` | string | LoggingTrait | Yes |
 | `metadata` | ObjectMeta | Refer to Kubernetes API documentation for fields of metadata. | No |
-| `spec` |  [LoggingTraitSpec](#loggingtraitspec) | The desired state of an ingress trait. | Yes |
+| `spec` |  [LoggingTraitSpec](#loggingtraitspec) | The desired state of an logging trait. | Yes |
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/OAM/LoggingTrait.md:65]: an logging trait -> a logging trait

File: content/en/docs/reference/API/OAM/LoggingTrait.md [hunk 7]
```diff
@@ -62,10 +62,10 @@ In this example, the `logging-stdout` container will run the image given in the
...
 
 #### LoggingTraitSpec
-LoggingTraitSpec specifies the desired state of an ingress trait.
+LoggingTraitSpec specifies the desired state of an logging trait.
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/OAM/LoggingTrait.md:68]: an logging trait -> a logging trait

**Review by rosemarymarano — COMMENTED** [review]
few more

File: content/en/docs/reference/API/OAM/LoggingTrait.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+| `kind` | string | LoggingTrait | Yes |
+| `metadata` | ObjectMeta | Refer to Kubernetes API documentation for fields of metadata. | No |
+| `spec` |  [LoggingTraitSpec](#loggingtraitspec) | The desired state of an ingress trait. | Yes |
+
+#### LoggingTraitSpec
+LoggingTraitSpec specifies the desired state of an ingress trait.
+
+| Field | Type | Description | Required
+| --- | --- | --- | --- |
+| `loggingConfig` | map[string]string | A  of the Fluentd configuration file name and the configuration file details. | Yes |
```
[the remark this answers is not in the document]
Comment (davisgomes) [reply content/en/docs/reference/API/OAM/LoggingTrait.md:72]: meant to put 'map' there

File: content/en/docs/reference/API/OAM/LoggingTrait.md (hunk 3 above)
Comment (davisgomes) [reply content/en/docs/reference/API/OAM/LoggingTrait.md:75]: Copy and paste error.

File: content/en/docs/reference/API/OAM/LoggingTrait.md [hunk 9]
```diff
@@ -62,22 +62,22 @@ In this example, the `logging-stdout` container will run the image given in the
...
-LoggingTraitSpec specifies the desired state of an logging trait.
+LoggingTraitSpec specifies the desired state of a logging trait.
 
 | Field | Type | Description | Required
 | --- | --- | --- | --- |
-| `loggingConfig` | map[string]string | A  of the Fluentd configuration file name and the configuration file details. | Yes |
+| `loggingConfig` | map[string]string | A map of the Fluentd configuration file name and the configuration file details. | Yes |
 | `loggingImage` | string | The name of the custom Fluentd image. | Yes |
-| `workloadRef` | [WorkloadReference](#workloadReference) | The name of the custom Fluentd image. | Yes |
+| `workloadRef` | [WorkloadReference](#workloadReference) | A reference to the desired workload | Yes |
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/OAM/LoggingTrait.md:74]: Period at the end of the sentence please.

**Review by rosemarymarano — COMMENTED** [review]
One last missing period. otherwise ok.

**Review by rosemarymarano — APPROVED** [review]

**Pull request merged**